src/operator/tensor/diag_op-inl.h

```diff
@@ -133,13 +167,94 @@ struct diag_gen {
     auto j = unravel(i, oshape);
     if (j[1] == (j[0] + k)) {
```

Comment:
if the input shape is (3,3) then the output diagonal will be (3), will this cause a problem with indexing j[1] since there wont be a 2nd dimension in the oshape?

Reply:
`diag_gen` is what is reserved for generating 2D matrices from 1D inputs (a functionality from the original `diag` operator). It would not be invoked when the input has more than 1 dimension.